package NYT.NRpc.NProto;

import "yt/core/misc/guid.proto";
import "yt/core/misc/error.proto";

////////////////////////////////////////////////////////////////////////////////

message TRequestHeader
{
    optional NYT.NProto.TGuid request_id = 1;
    required string service = 2;
    required string method = 3;
    optional NYT.NProto.TGuid realm_id = 8;

    optional bool one_way = 4;
    optional int64 request_start_time = 6;
    optional int64 retry_start_time = 7;

    extensions 100 to max;

}

////////////////////////////////////////////////////////////////////////////////

message TAuthenticatedExt
{
    extend NRpc.NProto.TRequestHeader
    {
        optional string authenticated_user = 102;
    }
}

message TResponseHeader
{
    optional NYT.NProto.TGuid request_id = 1;
    required NYT.NProto.TError error = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReqDiscover
{ }

message TRspDiscover
{
    required bool up = 1;
    repeated string suggested_addresses = 2;
}

////////////////////////////////////////////////////////////////////////////////

